packages/eslint-plugin/src/rules/use-unknown-in-catch-callback-variable.ts

```diff
@@ -235,69 +237,65 @@ export default createRule<[], MessageIds>({
 
     return {
       CallExpression({ arguments: args, callee }): void {
-        const promiseMethodInfo =
-          callee.type === AST_NODE_TYPES.MemberExpression &&
-          tsutils.isThenableType(
-            checker,
-            services.esTreeNodeToTSNodeMap.get(callee),
-            checker.getTypeAtLocation(
-              services.esTreeNodeToTSNodeMap.get(callee.object),
-            ),
-          ) &&
-          [
-            ...[
-              { method: 'catch', append: '' }, // argumentIndexToCheck: 0
-              { method: 'then', append: ' rejection' }, // argumentIndexToCheck: 1
-            ].entries(),
-          ].find(([, { method }]) =>
-            isStaticMemberAccessOfValue(callee, method),
-          );
+        if (callee.type !== AST_NODE_TYPES.MemberExpression) {
+          return;
+        }
+
+        const staticMemberAccessKey = getStaticMemberAccessKey(callee);
+        if (!staticMemberAccessKey) {
+          return;
+        }
+
+        const promiseMethodInfo = [
```

Comment:
make a type for this array - the trouble with inferring it is that one can omit `method` or `append`, or add extraneous properties, and the type checker won't complain. 

maybe like so?
```ts
type PromiseMethodInfo = { reportMessageData: { method: string, append: string }, argIndexToCheck: string }
```